src/components/common/Actions.vue

```diff
@@ -171,19 +171,19 @@ export default class Actions extends Mixins(DateMixin, FilingTemplateMixin, Name
     const paymentToken = filingComplete?.header?.paymentToken
     if (paymentToken) {
       const isPaymentActionRequired: boolean = filingComplete.header?.isPaymentActionRequired
-      const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
+      const returnUrl = sessionStorage.getItem('DASHBOARD_URL') + this.getBusinessId +
+        `?filing_id=${this.getFilingId}`
 
       // if payment action is required, navigate to Pay URL
       if (isPaymentActionRequired) {
         const authUrl = sessionStorage.getItem('AUTH_WEB_URL')
-        const returnUrl = encodeURIComponent(dashboardUrl + this.getBusinessId)
-        const payUrl = authUrl + 'makepayment/' + paymentToken + '/' + returnUrl
+        const payUrl = authUrl + 'makepayment/' + paymentToken + '/' + encodeURIComponent(returnUrl)
         // assume Pay URL is always reachable
         // otherwise user will have to retry payment later
         navigate(payUrl)
       } else {
         // otherwise go straight to dashboard
-        this.$root.$emit('go-to-dashboard')
+        navigate(returnUrl)
```

Comment:
This chunk of code is used for correction filings (which use the old Actions component).